TombEngine/Shaders/Blending.hlsli

```diff
@@ -24,182 +24,279 @@
+    noiseValue2 = AnimatedNebula(uv84, timeUniform * 0.1f);
 
-	frequency = 2.5;
-	amplitude = 0.2;
-	persistence = 4.7;
+    frequency = 2.5;
+    amplitude = 0.2;
+    persistence = 4.7;
 
-	float2 uv83 = uv * 8;
-	uv83.y = (uv.y + (timeUniform * 0.02));
-	noiseValue3 = NebularNoise(uv83, frequency, amplitude, persistence);
+    float2 uv83 = uv * 8;
+    uv83.y = (uv.y + (timeUniform * 0.02));
+    noiseValue3 = NebularNoise(uv83, frequency, amplitude, persistence);
 
-	noiseValue2 += AnimatedNebula(uv/2, timeUniform * 0.05f);
-	
-	color.rgb *= noiseValue2 + 0.6f;
-	color.rgb += noiseValue3;
-	color.a *= noiseValue + 0.01f;
+    noiseValue2 += AnimatedNebula(uv / 2, timeUniform * 0.05f);
 
-	color.rgb -= shadowx + 0.1f;
+    color.rgb *= noiseValue2 + 0.6f;
+    color.rgb += noiseValue3;
+    color.a *= noiseValue + 0.01f;
 
-	color.a *= noiseValue2 + 0.9f;
-	color.a *= noiseValue3 + 2.0f;
+    color.rgb -= shadowx + 0.1f;
 
-	float fade0 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradL, gradT)));
-	float fade1 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradL, gradB)));
-	float fade2 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradR, gradB)));
-	float fade3 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradR, gradT)));
+    color.a *= noiseValue2 + 0.9f;
+    color.a *= noiseValue3 + 2.0f;
 
-	float fadeL = 1.40f * faceFactor * faceFactor * (1.0 - gradL);
-	float fadeB = 2.75f * faceFactor * faceFactor * (1.0 - gradB);
-	float fadeR = 1.40f * faceFactor * faceFactor * (1.0 - gradR);
-	float fadeT = 2.75f * faceFactor * faceFactor * (1.0 - gradT);
+    float fade0 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradL, gradT)));
+    float fade1 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradL, gradB)));
+    float fade2 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradR, gradB)));
+    float fade3 = faceFactor * max(0.0, 1.0 - dot(float2(BLENDING, BLENDING), float2(gradR, gradT)));
 
-	float fade = max(
+    float fadeL = 1.40f * faceFactor * faceFactor * (1.0 - gradL);
+    float fadeB = 2.75f * faceFactor * faceFactor * (1.0 - gradB);
+    float fadeR = 1.40f * faceFactor * faceFactor * (1.0 - gradR);
+    float fadeT = 2.75f * faceFactor * faceFactor * (1.0 - gradT);
+
+    float fade = max(
+		max(max(fade0, fade1), max(fade2, fade3)),
+		max(max(fadeL, fadeR), max(fadeB, fadeT)));
+
+    float scale = 1.0 - fade;
+
+    color *= scale;
+    float decayFactor = 1.0f;
+    if (uv.y > 0.5f && uv.y < 1.0f)
+    {
+        decayFactor = uv.y / 2;
+    }
+    if (uv.y < 0.5f && uv.y > 0.0f)
+    {
+        decayFactor = (1.0f - uv.y) / 2;
+    }
+    color *= decayFactor;
+
+    color.rgb = smoothstep(ZERO, EIGHT_FIVE, color.rgb);
+    return color;
+}
+
+float4 DoLaserBeamEffect(float3 input, float4 output, float2 uv, float faceFactor, float timeUniform)

```

Comment:
`Blending.hlsli` becomes a "monster file" with all new sprite render types. Can we maybe move custom drawing shaders to a new file?

Reply:
Sorry I do not have the skills for it :(

someone else has to